lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/EvaluatorRuntime.cs

```diff
@@ -199,27 +202,30 @@ public void OnNext(REEFMessage value)
             }
         }
 
-        private void Handle(Exception e)
+        private void OnException(Exception e)
         {
             lock (_heartBeatManager)
             {
-                Logger.Log(Level.Error, string.Format(CultureInfo.InvariantCulture, "evaluator {0} failed with exception", _evaluatorId), e);
+                Logger.Log(Level.Error, "evaluator {0} failed with exception {1}.", _evaluatorId, e);
                 _state = State.FAILED;
-                string errorMessage = string.Format(
+
+                var errorMessage = string.Format(
                         CultureInfo.InvariantCulture,
                         "failed with error [{0}] with message [{1}] and stack trace [{2}]",
                         e,
                         e.Message,
                         e.StackTrace);
-                EvaluatorStatusProto evaluatorStatusProto = new EvaluatorStatusProto()
+
+                var evaluatorStatusProto = new EvaluatorStatusProto()
                 {
                     evaluator_id = _evaluatorId,
                     error = ByteUtilities.StringToByteArrays(errorMessage),
                     state = _state
                 };
+
                 _heartBeatManager.OnNext(evaluatorStatusProto);
```

Comment:
Shouldn't that message also contain the IDs of failed contexts as well as the ID of the failed task (if any)?